client/src/profile/Addresses.js

```diff
@@ -0,0 +1,35 @@
+import React, { PropTypes } from 'react';
+import AddressField from './AddressField';
+
+const Addresses = ({
+  addresses,
+  saveNewAddress,
+  updateAddressState,
+  addressFieldState,
+  removeAddress,
+ }) => (
+   <div>
+     <h3>Receive Alerts For Which Addresses?</h3>
+     {addresses.map(address => (
+       <div key={address.address}>
```

Comment:
This would be a good section to pull out for its own component. Just a suggestion though, no need to do it now :)